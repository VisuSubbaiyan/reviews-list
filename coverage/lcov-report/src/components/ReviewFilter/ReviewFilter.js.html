<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/ReviewFilter/ReviewFilter.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/ReviewFilter</a> ReviewFilter.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">85.71% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">16x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import { debounce } from 'lodash';
&nbsp;
import { ReviewFilterContainer, StyledReviewFilter } from './ReviewFilter.styled';
import { TextInput, Select } from '../base';
import { groupByOptions, sortByOptions, filterOptions } from '../../helpers/util';
&nbsp;
const ReviewFilter = ({ onChange, search, groupBy, sortBy, stars, diableSort }) =&gt; {
  const handleOnChange = name =&gt; <span class="fstat-no" title="function not covered" >va</span>lue =&gt; {
<span class="cstat-no" title="statement not covered" >    onChange({ [name]: name === 'stars' ? Number(value) : value });</span>
  };
&nbsp;
  const css = `width: 32%; margin: 7px 0;`;
  return (
    &lt;ReviewFilterContainer&gt;
      &lt;StyledReviewFilter&gt;
        &lt;TextInput
          onChange={debounce(handleOnChange('search'), 300)}
          value={search}
          placeholder="Search reviews"
          css={`margin: 7px 0; padding: 5px 10px;`}
        /&gt;
        &lt;Select
          onChange={handleOnChange('groupBy')}
          options={groupByOptions}
          selected={groupBy}
          css={css}
        /&gt;
        &lt;Select
          onChange={handleOnChange('stars')}
          options={filterOptions}
          selected={stars}
          css={css}
        /&gt;
        &lt;Select
          onChange={handleOnChange('sortBy')}
          options={sortByOptions}
          selected={sortBy}
          disabled={diableSort}
          css={css}
        /&gt;
      &lt;/StyledReviewFilter&gt;
    &lt;/ReviewFilterContainer&gt;
  );
};
&nbsp;
ReviewFilter.propTypes = {
  onChange: PropTypes.func.isRequired,
  search: PropTypes.string,
  groupBy: PropTypes.string.isRequired,
  sortBy: PropTypes.string.isRequired,
  diableSort: PropTypes.bool.isRequired
};
&nbsp;
export default React.memo(ReviewFilter);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon May 27 2019 19:38:35 GMT+0200 (CEST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
