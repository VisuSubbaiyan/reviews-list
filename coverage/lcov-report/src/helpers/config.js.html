<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/helpers/config.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/helpers</a> config.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/1</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20</td><td class="line-coverage quiet"><span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export const theme = {
  primary: '#20232a',
  secondary: '#ccc',
  black: '#282c34',
  mediumGrey: '#aaa',
  lightGrey: '#dcdcdc',
  mediumFont: '14px',
  smallFont: '10px',
  linkColor: '#0366d6',
  blue: '#39a2d6',
  blueGrey: '#7998a8',
  formControlRadius: '5px',
  boxRadius: '10px',
  buttonRadius: '15px',
  breakpoints: {
    small: 640,
    medium: 769
  }
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon May 27 2019 19:38:35 GMT+0200 (CEST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
