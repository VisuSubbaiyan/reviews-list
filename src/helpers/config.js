export const theme = {
  primary: '#20232a',
  secondary: '#ccc',
  black: '#282c34',
  mediumGrey: '#aaa',
  lightGrey: '#dcdcdc',
  mediumFont: '14px',
  smallFont: '10px',
  linkColor: '#0366d6',
  blue: '#39a2d6',
  blueGrey: '#7998a8',
  formControlRadius: '5px',
  boxRadius: '10px',
  buttonRadius: '15px',
  breakpoints: {
    small: 640,
    medium: 769
  }
};
