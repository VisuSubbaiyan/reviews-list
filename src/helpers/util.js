import { css } from 'styled-components';
import { format as formatDate, endOfWeek, startOfWeek, startOfMonth } from 'date-fns';

/** Styled compoenent uitls */
// Based on the number of codes we can move this to seperate file
export const styling = props => props.css && css`${props.css}`;

export const viewport = breakpoint => (...args) => css`
    @media (min-width: ${breakpoint}px) {
      ${css(...args)}
    }
  `;

export const iconSizes = {
  xs: '12px',
  s: '16px',
  m: '24px',
  l: '32px',
  xl: '38px',
};

export const getSize = props => iconSizes[props.size];


/** Reviews component utils */
export const formatTitle = (title, group) => {
  switch (group) {
    case 'week':
      return `${formatDate(title, 'DD.MM')} - ${formatDate(endOfWeek(title), 'DD.MM.YYYY')}`;
    case 'month':
      return formatDate(title, 'MMMM YYYY');
    default:
      return formatDate(title, 'DD MMMM YYYY');
  }
};

export const isTextMatch = (text, match) => {
  const textLowerCase = text.toLowerCase();
  const matchLowerCase = match.toLowerCase();

  return textLowerCase.includes(matchLowerCase);
};

export const isReviewMatch = (review, matchWith) => {
  const { title, content } = review;

  return matchWith && (isTextMatch(title, matchWith) || isTextMatch(content, matchWith));
};

export const groupByIterator = group => review => {
  // GroupBy
  // Day new Date();
  // Week startOfWeek(new Date());
  // Month startOfMonth(new Date());
  const reviewedDate = new Date(review.reviewCreated);

  switch (group) {
    case 'week': {
      return startOfWeek(reviewedDate);
    }
    case 'month': {
      return startOfMonth(reviewedDate);
    }
    default: {
      return reviewedDate;
    }
  }
}

export const sortByOptions = [{ value: 'desc', name: 'Most Recent' }, { value: 'asc', name: 'Older' }];

export const groupByOptions = [
  { value: 'day', name: 'Group by day' },
  { value: 'week', name: 'Group by week' },
  { value: 'month', name: 'Group by month' }
];

export const filterOptions = [
  { value: 5, name: '5 star only' },
  { value: 4, name: '4 star only' },
  { value: 3, name: '3 star only' },
  { value: 2, name: '2 star only' },
  { value: 1, name: '1 star only' }
];

export const smoothScrollToPosition = top => window.scrollTo({
    top,
    behavior: 'smooth',
  });
