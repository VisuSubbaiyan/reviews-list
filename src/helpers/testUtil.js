import React from 'react';
import { mount, shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { prettyPrint } from 'html';

import { theme } from './config';
import { UPDATE_LOADING } from '../redux/actions';

export const createMountedComponent = ({ children, isShallow }) => {
  const component = (
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  );
  const mountedComponent = isShallow ? shallow(component) : mount(component);
  mountedComponent.htmlRender = () => prettyPrint(mountedComponent.html(), { indent_size: 2 });

  return mountedComponent;
};

export const sampleText = 'sample text';
export const mockFunction = jest.fn(value => value);

export const inputProps = {
  value: 'sample name',
  onChange: mockFunction,
  placeholder: 'sample placeholder'
};

export const sampleState = {
  search: '',
  stars: 5,
  groupBy: 'day',
  sortBy: 'desc',
  diableSort: false
};

export const updateLoadingAction = {
  type: UPDATE_LOADING,
  payload: { loading: true }
};

export const expectedLoadingState = {
  page: 1,
  loading: true,
  reviews: [],
  hasMore: false
}
