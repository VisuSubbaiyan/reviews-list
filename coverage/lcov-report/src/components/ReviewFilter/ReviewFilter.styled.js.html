<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/ReviewFilter/ReviewFilter.styled.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/ReviewFilter</a> ReviewFilter.styled.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import styled from 'styled-components';
import { viewport } from '../../helpers/util';
&nbsp;
import { GeneralContiner } from '../base';
&nbsp;
export const StyledReviewFilter = styled.div`
  display: flex;
  top: 60px;
  flex-flow: row wrap;
  justify-content: space-between;
  min-height: 60px;
  width: 95%;
  /* background: ${props =&gt; props.theme.lightGrey} */
&nbsp;
  ${props =&gt; viewport(props.theme.breakpoints.small)`width: 720px;`}
`;
&nbsp;
export const ReviewFilterContainer = styled(GeneralContiner)`
    width: 100vw;
    background: ${props =&gt; props.theme.mediumGrey};
    position: fixed;
    z-index: 1;
    left: 0;
    top: 60px;
`;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon May 27 2019 19:38:35 GMT+0200 (CEST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
