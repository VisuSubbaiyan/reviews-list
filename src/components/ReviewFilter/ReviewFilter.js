import React from 'react';
import PropTypes from 'prop-types';
import { debounce } from 'lodash';

import { ReviewFilterContainer, StyledReviewFilter } from './ReviewFilter.styled';
import { TextInput, Select } from '../base';
import { groupByOptions, sortByOptions, filterOptions } from '../../helpers/util';

const ReviewFilter = ({ onChange, search, groupBy, sortBy, stars, diableSort }) => {
  const handleOnChange = name => value => {
    onChange({ [name]: name === 'stars' ? Number(value) : value });
  };

  const css = `width: 32%; margin: 7px 0;`;
  return (
    <ReviewFilterContainer>
      <StyledReviewFilter>
        <TextInput
          onChange={debounce(handleOnChange('search'), 300)}
          value={search}
          placeholder="Search reviews"
          css={`margin: 7px 0; padding: 5px 10px;`}
        />
        <Select
          onChange={handleOnChange('groupBy')}
          options={groupByOptions}
          selected={groupBy}
          css={css}
        />
        <Select
          onChange={handleOnChange('stars')}
          options={filterOptions}
          selected={stars}
          css={css}
        />
        <Select
          onChange={handleOnChange('sortBy')}
          options={sortByOptions}
          selected={sortBy}
          disabled={diableSort}
          css={css}
        />
      </StyledReviewFilter>
    </ReviewFilterContainer>
  );
};

ReviewFilter.propTypes = {
  onChange: PropTypes.func.isRequired,
  search: PropTypes.string,
  groupBy: PropTypes.string.isRequired,
  sortBy: PropTypes.string.isRequired,
  diableSort: PropTypes.bool.isRequired
};

export default React.memo(ReviewFilter);
