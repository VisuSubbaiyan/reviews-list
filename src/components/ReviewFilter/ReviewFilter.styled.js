import styled from 'styled-components';
import { viewport } from '../../helpers/util';

import { GeneralContiner } from '../base';

export const StyledReviewFilter = styled.div`
  display: flex;
  top: 60px;
  flex-flow: row wrap;
  justify-content: space-between;
  min-height: 60px;
  width: 95%;
  /* background: ${props => props.theme.lightGrey} */

  ${props => viewport(props.theme.breakpoints.small)`width: 720px;`}
`;

export const ReviewFilterContainer = styled(GeneralContiner)`
    width: 100vw;
    background: ${props => props.theme.mediumGrey};
    position: fixed;
    z-index: 1;
    left: 0;
    top: 60px;
`;
