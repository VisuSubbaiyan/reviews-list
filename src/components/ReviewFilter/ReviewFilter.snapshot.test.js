import React from 'react';
import ReviewFilter from './ReviewFilter';
import { createMountedComponent, sampleState, mockFunction } from '../../helpers/testUtil';

describe('ReviewFilter should', () => {
  xit('render correctly according to props', () => {
    const mountedReviewFilter = createMountedComponent({
      children: <ReviewFilter {...sampleState} onChange={mockFunction} />
    });

    expect(mountedReviewFilter.htmlRender()).toMatchSnapshot();
    expect(mountedReviewFilter.find('input').length).toEqual(1);
    expect(mountedReviewFilter.find('select').length).toEqual(3);
  });
});