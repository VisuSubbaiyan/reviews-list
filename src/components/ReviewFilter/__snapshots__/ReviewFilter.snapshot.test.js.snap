// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewFilter should render correctly according to props 1`] = `
"<div class=\\"sc-gzVnrw sc-jTzLTM kVNAQh\\">
  <div class=\\"sc-VigVT gFzrRt\\">
    <input placeholder=\\"Search reviews\\" class=\\"sc-iwsKbI bwkJWA\\" value=\\"\\">
    <select class=\\"sc-gqjmRU cxVyKR\\">
      <option value=\\"day\\">Group by day</option>
      <option value=\\"week\\">Group by week</option>
      <option value=\\"month\\">Group by month</option>
    </select>
    <select class=\\"sc-gqjmRU cxVyKR\\">
      <option value=\\"5\\">5 star only</option>
      <option value=\\"4\\">4 star only</option>
      <option value=\\"3\\">3 star only</option>
      <option value=\\"2\\">2 star only</option>
      <option value=\\"1\\">1 star only</option>
    </select>
    <select class=\\"sc-gqjmRU cxVyKR\\">
      <option value=\\"desc\\">Most Recent</option>
      <option value=\\"asc\\">Older</option>
    </select>
  </div>
</div>"
`;
